src/commands/registers.ts

```diff
@@ -222,10 +234,386 @@ export class SomeRegisterCommand extends EmacsCommand {
+      this.registers.set(registerKey, { type: "rectangle", rectTexts: copiedRectTexts });
     }
 
     this.emacsController.exitMarkMode();
     makeSelectionsEmpty(textEditor);
   }
+
+  // point-to-register, C-x r SPC <r>
+  public async runPoint(textEditor: vscode.TextEditor, registerKey: string): Promise<void> {
+    // Save all cursor positions as new Position objects to ensure immutability
+    const positions = textEditor.selections.map(
+      (selection) => new vscode.Position(selection.active.line, selection.active.character),
+    );
+    // Store positions with explicit type annotation
+    if (positions.length === 0) {
+      MessageManager.showMessage("No position to store");
+      return;
+    }
+    // Ensure all positions are valid VSCode Position objects
+    const validPositions = positions.filter((pos): pos is vscode.Position => pos instanceof vscode.Position);
+    if (validPositions.length === 0) {
+      MessageManager.showMessage("Invalid positions");
+      return;
+    }
+    const registerData: PositionRegisterData = {
+      type: "positions",
+      positions: validPositions,
+    };
+    this.registers.set(registerKey, registerData);
+    // Keep selections empty at current positions
+    makeSelectionsEmpty(textEditor);
+    MessageManager.showMessage("Position saved in register");
+  }
+
+  // jump-to-register, C-x r j <r>
+  public async runJump(textEditor: vscode.TextEditor, registerKey: string): Promise<void> {
+    if (!this.registers.has(registerKey)) {
+      MessageManager.showMessage("Register does not contain a position");
+      return;
+    }
+
+    const data = this.registers.get(registerKey);
+    if (data?.type !== "positions") {
+      MessageManager.showMessage("Register does not contain a position");
+      return;
+    }
+
+    // Get and validate current positions
+    const currentPositions = textEditor.selections.map((s) => s.active);
+    if (currentPositions.length === 0) {
+      MessageManager.showMessage("No valid current positions");
+      return;
+    }
+
+    if (!data.positions || data.positions.length === 0) {
+      MessageManager.showMessage("No position stored in this register");
+      return;
+    }
+
+    // Get positions from register and ensure they are valid
+    const storedPositions = data.positions;
+    if (!storedPositions || storedPositions.length === 0) {
+      MessageManager.showMessage("No positions stored in register");
+      return;
+    }
+
+    // First, save current positions as marks before moving
+    const currentMarks = currentPositions.map((pos) => new vscode.Position(pos.line, pos.character));
+
+    // Ensure we have at least one mark
+    if (currentMarks.length === 0) {
+      return;
+    }
+
+    // Create selections using the stored positions, ensuring type safety
+    const validStoredPositions = storedPositions.filter((pos): pos is vscode.Position => {
```

Comment:
Is this runtime check actually needed?
These values should already been type-safe.